File: backend/app/schemas/entry.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class EntryStart(BaseModel):
    label: str


class EntryUpdate(BaseModel):
    label: Optional[str] = None
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None


class EntryOut(BaseModel):
    id: int
    user_id: int
    label: str
    started_at: datetime
    ended_at: Optional[datetime] = None
    duration_seconds: Optional[int] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: backend/src/schemas/entry.ts
import { z } from "zod";

export const entryStartSchema = z.object({
  label: z.string().min(1),
});

export const entryUpdateSchema = z.object({
  label: z.string().optional(),
  started_at: z.string().optional(),
  ended_at: z.string().nullable().optional(),
});

export type EntryStart = z.infer<typeof entryStartSchema>;
export type EntryUpdate = z.infer<typeof entryUpdateSchema>;

export interface EntryOut {
  id: number;
  user_id: number;
  label: string;
  started_at: string;
  ended_at: string | null;
  duration_seconds: number | null;
  created_at: string;
}